src/main/java/uk/nhs/hee/tis/trainee/reference/api/GradeResource.java

```diff
@@ -86,6 +95,12 @@ public GradeResource(GradeService service, GradeMapper mapper) {
    */
   @PutMapping("/grade")
   public ResponseEntity<GradeDto> updateGrade(@RequestBody GradeDto gradeDto) {
+    if (!validator.isValid(gradeDto)) {
+      // TODO: refactor service to take DTO instead of entity and move business logic there?
+      service.deleteByTisId(gradeDto.getTisId());
```

Comment:
I'm sure there's a good reason for this, but isn't it a little surprising to delete an existing entry from the database if an invalid edit is posted? I might have thought we'd leave the existing record in the database as-is

Reply:
Calling it "valid" isn't truly accurate I guess.

As this PR stands any grade that we don't want in the trainee reference table is flagged as "invalid". The reason it is deleted is we don't want to leave historical Grade behind when they've been updated to an invalid state within TIS.

Reply:
Ok, that makes sense, thanks for clarifying!